guides/partials.md

```diff
@@ -1,36 +1,54 @@
+Partials rely on the JSON context of the parent template invoking it. Like [sections](/guides/getting-started/#sections), partials also accept parameters that add extra references. By passing more data into the partial using parameters, you can build components that are customized to different contexts easily. For example, there might be several versions of a header that you wish to include, which can be controlled with code like `{>"header" mode="classic"/}`.
 
-Just like in sections, inline parameters will not override the current context if a property of the same name exists. For example, if the current context already has {name: "Albert"} adding name as a parameter will not override the value when used inside the partial foo.
+Just like in sections, inline parameters will not override the current context if a reference of the same name already exists.
 
-```{>foo name="will not override Albert"/}```
+In the following example, notice that `isGreeting` is only defined in the data for the `disney` template, but it is available in every instance of the included `disney-park` partial, while `parkName` and `qualifier` vary per instance.
 
-You will use parameters to pass an object like:
-
-<dust-demo template-name="list_item">
-<dust-demo-template><li>{name}</li></dust-demo-template>
-<dust-demo-json>{}</dust-demo-json>
+<dust-demo templateName="disney-park" hideOutput="true">
+<dust-demo-template showTemplateName="true">
+<p>{?isGreeting}Greetings{:else}Goodbye{/isGreeting} from {parkName}, The {qualifier} Place on Earth!</p>
+</dust-demo-template>
 </dust-demo>
-<dust-demo template-name="list">
-  <dust-demo-template><ul>{#names}{>"list_item" name=name/}{/names}</ul></dust-demo-template>
-  <dust-demo-json>
+
+<dust-demo templateName="disney">
+<dust-demo-template showTemplateName="true">
+{#parks}
+  {>"disney-park" parkName=name qualifier=qualifier/} 
+{/parks}
+</dust-demo-template>
+<dust-demo-json>
+{
+  "isGreeting": true,
+  "parks": [
+    {
+      "name": "Disneyland",
+      "qualifier": "Happiest"
+    },
     {
-      names: ['Dust', 'partial', 'example']
+      "name": "Disney World Magic Kingdom",
+      "qualifier": "Most Magical"
     }
-  </dust-demo-json>
-
+  ]
+}
+</dust-demo-json>
 </dust-demo>
 
+Let's see how the Dust template rendering knows about a template. Since Dust templates are compiled to JavaScript, part of that compiled result is a call to `dust.register(name, functionImplementingCompiledTemplate)`---check out the Dev Mode of the example above to see this.
+
+The name is not necessarily the file name, but it will be the template name that is used when including the partial with the include syntax. This means that it doesn't really matter what extension you give your Dust template files, though `.dust` and `.tl` are most commonly seen.
```

Comment:
I wanted to keep some of the text about what the template name is when you include the file, because it seems unobvious to me that it's not necessarily the file name. Should this go somewhere else maybe?